### PR TITLE
Feature/dev 614 sell dex fee note

### PR DESCRIPTION
**Pull request by ThaBrad**
#### What kind of PR is this?:
<!-- Use one of the following kinds:
/kind feature

- [ ] Read your code changes at least once
- [ ] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [ ] No console errors on web
- [ ] Tested on Light mode and Dark mode*
- [ ] Your UI implementation visually matched the rendered design*
- [ ] Added translations*

### CODE REVIEW

**Comment by ThaBrad** [comment]
only waiting for @OliLemmur to finish translation

File: mobile-app/app/screens/AppNavigator/screens/Portfolio/screens/SellScreen.tsx [hunk 1]
```diff
@@ -318,6 +319,12 @@ export function SellScreen ({
           )
           : (
             <>
+              {(token.displaySymbol !== 'DFI' && token.displaySymbol !== 'dBTC' && token.displaySymbol !== 'dETH' && token.displaySymbol !== 'dUSDT' && token.displaySymbol !== 'dUSDC') && (
```
[the remark this answers is not in the document]
Comment (ThaBrad) [reply mobile-app/app/screens/AppNavigator/screens/Portfolio/screens/SellScreen.tsx:322]: > What about dDoge, dBCH and dLTC?

I only added the first couple from the text for convenience. Was not a must requirement.


> Maybe use `!['DFI', 'dBTC', 'dETH', ...].includes(token.displaySymbol) && ...`

Adds additional execution runtime and doesn't even interrupt early on truthiness 🙁 


> Or even better, use the `useDexStabilization(tokenToSell, DFI)` hook and only show the message, if the stabilization type is not 'none'?

good catch!

[the remark this answers is not in the document]
Comment (davidleomay) [reply mobile-app/app/screens/AppNavigator/screens/Portfolio/screens/SellScreen.tsx:322]: > Adds additional execution runtime and doesn't even interrupt early on truthiness 🙁

But it makes it much more readable 😉  and are you sure, that `includes` does not return on match? Would be a quite stupid implementation ...

[the remark this answers is not in the document]
Comment (ThaBrad) [reply mobile-app/app/screens/AppNavigator/screens/Portfolio/screens/SellScreen.tsx:322]: Yeah… if it gets longer.

But no, includes apparently doesn't short-circuit => https://stackoverflow.com/questions/8710442/how-to-specify-multiple-conditions-in-an-if-statement-in-javascript#comment129367715_51565021

Although there are conflicting statements: 
https://stackoverflow.com/questions/51711704/using-array-includes-instead-of-or-chain#comment90385081_51711824
It seems it short-circuits on the function, but the Array still has to be completely evaluated => so only kinda half short-circuits, not like && or ||. 
BUT if typescript, or whatever comes next, optimises ".includes" to a short-circuiting version of the code, it wouldn't matter anyways 🤷‍♂️ 

Nerd-deep-dive over …🤣

**Comment by ThaBrad** [comment]
Ich bau noch schnell das Dropdown für die Fee's…

File: mobile-app/app/screens/AppNavigator/screens/Portfolio/screens/SellScreen.tsx [hunk 2]
```diff
@@ -318,6 +320,17 @@ export function SellScreen ({
...
           : (
             <>
+              {token.symbol !== 'DFI' && (
+                <ThemedView style={tailwind('px-4 mb-4')}>
+                  <DfxDexFeeInfo
+                    token={token}
+                    getDexFee={(df) => setTimeout(() => {
+                      setDexFee(df)
+                    }, 50)}
+                  />
```
Comment (davidleomay) [inline mobile-app/app/screens/AppNavigator/screens/Portfolio/screens/SellScreen.tsx:330]: Interesting approach 🤔

File: mobile-app/app/screens/WalletNavigator/screens/RestoreWallet/RestoreMnemonicWallet.tsx [hunk 3]
```diff
@@ -25,18 +26,35 @@ export function RestoreMnemonicWallet (): JSX.Element {
...
+    })()
+  }
+
   useEffect(() => {
     recoveryWords.forEach((r) => {
       inputRefMap[r] = createRef<TextInput>()
       setInputRefMap(inputRefMap)
     })
+
+    // autoFillFromClipboard()
```
Comment (davidleomay) [inline mobile-app/app/screens/WalletNavigator/screens/RestoreWallet/RestoreMnemonicWallet.tsx:57]: Unused

File: website/src/pages/api/v0/settings/flags.ts [hunk 4]
```diff
@@ -84,12 +84,22 @@ export default async function handle (req: NextApiRequest, res: NextApiResponse<
...
       platforms: ['ios', 'android', 'web']
-    }, {
+    },
+    {
       id: 'dusd_dfi_high_fee',
       name: 'DUSD-DFI High Fees',
       stage: 'public',
       version: '>0.0.0',
-      description: 'There are high fees in DUSD and DFI as of the moment',
+      description: 'There are high fees in some pairs with DUSD as of the moment',
```
Comment (davidleomay) [inline website/src/pages/api/v0/settings/flags.ts:93]: Why the same description as below? Isn't this one for DUSD-DFI?

File: mobile-app/app/components/DfxDexFeeInfo.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    } else {
+      props.getDexFee?.(dexStabilizationFee)
+    }
+  }
+
+  returnDexFee()
+
+  useEffect(() => {
+    returnDexFee()
+  }, [props.token])
```
Comment (davidleomay) [inline mobile-app/app/components/DfxDexFeeInfo.tsx:71]: Is this really a good approach? 😉  Wouldn't it be easier to have a separate reusable hook for this?

File: mobile-app/app/components/DfxDexFeeInfo.tsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+  } = useDexStabilization(dusd, dfi)
+
+  const isCrypto = (): boolean => ['DFI', 'dBTC', 'dETH', 'dUSDT', 'dUSDC', 'dLTC', 'dBCH', 'dDOGE'].includes(props.token.displaySymbol)
+
+  const returnDexFee = (): void => {
+    if (isCrypto()) {
+      props.getDexFee?.('0')
+    } else {
+      props.getDexFee?.(dexStabilizationFee)
+    }
```
Comment (davidleomay) [inline mobile-app/app/components/DfxDexFeeInfo.tsx:64]:
```ts
props.getDexFee?.(isCrypto() ? '0' : dexStabilizationFee)
```

File: mobile-app/app/screens/AppNavigator/screens/Portfolio/screens/SellScreen.tsx (hunk 2 above)
Comment (ThaBrad) [reply mobile-app/app/screens/AppNavigator/screens/Portfolio/screens/SellScreen.tsx]: fixed, reworked bad pattern workaround from before

File: mobile-app/app/screens/WalletNavigator/screens/RestoreWallet/RestoreMnemonicWallet.tsx (hunk 3 above)
Comment (ThaBrad) [reply mobile-app/app/screens/WalletNavigator/screens/RestoreWallet/RestoreMnemonicWallet.tsx]: left for future use if feature is demanded more often

**Pull request merged**